Adds changelog for 12.1.3

gurkanindibay opened the pull request.

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+
+* Fix performance issue in GetForeignKeyOids on systems with many constraints (#7578)
+
+* Fix performance issue when creating distributed tables and many already exist (#7575)
+
+* Fix performance issue when creating distributed tables if many already exist (#7579)
```
let's combine them to one

```suggestion
* Fixes a performance issue when creating distributed tables if many already exist (#7575, #7579)
```

gurkanindibay replied:
Added as you suggested

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+
+* Correctly mark some variables as volatile (#7570)
+
+* Fix performance issue in GetForeignKeyOids on systems with many constraints (#7578)
```
Better to have a more user friendly description --users are not expected to know underlying C function names:

```suggestion
* Fixes performance issue in when tracking foreign key constraints on systems with many constraints (#7578)
```

gurkanindibay replied:
Added as you suggested

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+* Allows overwriting host name for all inter-node connections by supporting "host"
+  parameter in citus.node_conninfo (#7541)
+
+* Correctly mark some variables as volatile (#7570)
```
We use present simple tense to start description entries:

```suggestion
* Fixes some potential bugs by correctly marking some variables as volatile (#7570)
```

gurkanindibay replied:
Added as you suggested

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+
+* Fixes a crash caused by some form of ALTER TABLE ADD COLUMN statements.
+  When adding multiple columns, if one of the ADD COLUMN statements contains a FOREIGN
+  constraint ommitting the referenced columns in the statement, a SEGFAULT occurs. (#7522)
```
```suggestion
  constraint omitting the referenced columns in the statement, a SEGFAULT was occurring. (#7522)
```

gurkanindibay replied:
Added as you suggested

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+
+* Fixes invalid copy-paste rightComparisonLimit (#7547)
+
+* Log username in the failed connection message (#7432)
```
```suggestion
* Logs username in the failed connection message (#7432)
```

gurkanindibay replied:
Added as you suggested

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,28 @@
+  When adding multiple columns, if one of the ADD COLUMN statements contains a FOREIGN
+  constraint ommitting the referenced columns in the statement, a SEGFAULT occurs. (#7522)
+
+* Fixes invalid copy-paste rightComparisonLimit (#7547)
```
let's delete this because in-practice this doesn't cause user-facing issue: https://github.com/citusdata/citus/issues/7543#issuecomment-1975242612

gurkanindibay replied:
Deleted

onurctirtir reviewed (commented):
The following are missing changelog entries, could you work with @hanefi to come up with changelog entries for those PRs?

Let's also write down changelog entries you would come up with back to relevant PR descriptions so that in a future release we can re-use them without having to check changelog.md:
* https://github.com/citusdata/citus/pull/7554
* https://github.com/citusdata/citus/pull/7492
* https://github.com/citusdata/citus/pull/7542
* https://github.com/citusdata/citus/pull/7377 (hint: https://github.com/citusdata/citus/pull/7377#issuecomment-1877017306)

gurkanindibay commented:
> The following are missing changelog entries, could you work with @hanefi to come up with changelog entries for those PRs?
> 
> Let's also write down changelog entries you would come up with back to relevant PR descriptions so that in a future release we can re-use them without having to check changelog.md:
> 
> * [Fix segfault. Issue #7381 #7554](https://github.com/citusdata/citus/pull/7554)
> * [Fix error in master_disable_node/citus_disable_node #7492](https://github.com/citusdata/citus/pull/7492)
> * [Change the order in which the locks are acquired #7542](https://github.com/citusdata/citus/pull/7542)
> * [fix multi_cluster_management check timeout  #7377](https://github.com/citusdata/citus/pull/7377) (hint: [fix multi_cluster_management check timeout  #7377 (comment)](https://github.com/citusdata/citus/pull/7377#issuecomment-1877017306))

Added descriptions in those PR's and added those descriptions as the changelog item in the Changelog file. @hanefi @onurctirtir  could you check if it is ok?

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+* Fixes bug when hostname in pg_dist_node resolves to multiple IPs
+  (#7377)
+
+* Fixes performance issue in when tracking foreign key constraints on
+  systems with many constraints (#7578)
```
```suggestion
* Fixes performance issue when tracking foreign key constraints on
  systems with many constraints (#7578)
```

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+  many tables (#7577)
+
+* Fixes a crash caused by some form of ALTER TABLE ADD COLUMN
+  statements.When adding multiple columns, if one of the ADD COLUMN
```
```suggestion
  statements. When adding multiple columns, if one of the ADD COLUMN
```

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+* Fixes performance issue in when tracking foreign key constraints on
+  systems with many constraints (#7578)
+
+* Fixes segmentation fault when using CASE WHEN in DO block functions
+  (#7554)
```
```suggestion
* Fixes segmentation fault when using CASE WHEN in DO block within
  functions. (#7554)
```

onurctirtir commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+* Fixes a performance issue when creating distributed tables if many
+  already exist (#7575, #7579)
+
+* Fixes bug when hostname in pg_dist_node resolves to multiple IPs
```
```suggestion
* Fixes a bug when hostname in pg_dist_node resolves to multiple IPs
```

onurctirtir commented:
(Sorry, I was out on Friday afternoon hence just had a chance to take a look into the PR)

hanefi commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+  reference tables, when a modify request is initiated from a worker
+  node that is not the "FirstWorkerNode" (#7542)
+
+* Fixes performance issue when distributing a table that depends on an
```
```suggestion
* Fixes a performance issue when distributing a table that depends on an
```

hanefi commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+* Fixes performance issue when distributing a table that depends on an
+  extension (#7574)
+
+* Fixes performance issue when using "\d tablename" on a server with
```
```suggestion
* Fixes a performance issue when using "\d tablename" on a server with
```

hanefi commented on CHANGELOG.md:
```diff
@@ -1,3 +1,43 @@
+* Fixes bug when hostname in pg_dist_node resolves to multiple IPs
+  (#7377)
+
+* Fixes performance issue in when tracking foreign key constraints on
```
```suggestion
* Fixes a performance issue in when tracking foreign key constraints on
```

hanefi reviewed (changes requested):
Please check line lengths after applying my suggestions.
